libr/anal/p/anal_gb.c

```diff
@@ -714,7 +714,10 @@ static bool gb_custom_daa (RAnalEsil *esil) {
 static int gb_anop(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *data, int len, RAnalOpMask mask) {
 	int ilen = gbOpLength (gb_op[data[0]].type);
 	if (ilen > len) {
-		ilen = 0;
+		op->addr = addr;
+		op->type = R_ANAL_OP_TYPE_UNK;
```

Comment:
i would say that if the instruction trying to be disassembled is bigger than the provided buffer the type should be INV (as for invalid) and about op->size. i would keep the original opsize, and just return 0 or -1. because this way the user knows how many bytes must be provided to get the disasm of that instruction.

also i think op->addr is filled by the caller of this callback. so no need to set it in here.

Reply:
pìng

Reply:
sorry for the delay, my laptop broke and i had to wait for spare parts to repair

Reply:
"also i think op->addr is filled by the caller of this callback. so no need to set it in here." - why do you pass addr to plugin->op then, if you already pass op to it?